File: Algorithms/Strings/Easy/Strong Password/Strong Password.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'minimumNumber' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts following parameters:
     *  1. INTEGER n
     *  2. STRING password
     */

    public static int minimumNumber(int n, string password)
    {
    // Return the minimum number of characters to make the password strong
    bool has_number=true,has_lower=true,has_upper=true,has_special=true;
    string special_characters = "!@#$%^&*()-+";
    int i=0;
    while( (i<n) && (has_number || has_lower || has_upper || has_special) ){
        if(password[i] >= 'a' && password[i] <= 'z')
            has_lower = false;
        else if(password[i] >= 'A' && password[i] <= 'Z'){
            has_upper = false;
        }
        else if(password[i] >= '0' && password[i] <= '9'){
            has_number = false;
        }
        else{
            for(int j=0;j<special_characters.Length;j++){
                if(special_characters[j]==password[i]){
                    has_special = false;
                    break;
                }
            }
        }
        i++;
    }
    int ans = 0;
    if(has_lower)
        ans++;
    if(has_upper)
        ans++;
    if(has_number)
        ans++;
    if(has_special)
        ans++;
    if(n<6){
        ans = Math.Max(ans, 6 - n);
    }
    return ans;
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        string password = Console.ReadLine();

        int answer = Result.minimumNumber(n, password);

        textWriter.WriteLine(answer);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Algorithms/Strings/Easy/Strong Password/Strong Password.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'minimumNumber' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER n
#  2. STRING password
#

def minimumNumber(n, password):
    # Return the minimum number of characters to make the password strong
    has_number = True
    has_lower = True
    has_upper = True
    has_special = True
    special_characters = "!@#$%^&*()-+"
    i=0
    while( (i<n) and (has_number or has_lower or has_upper or has_special) ):
        if(password[i] >= 'a' and password[i] <= 'z'):
            has_lower = False
        elif(password[i] >= 'A' and password[i] <= 'Z'):
            has_upper = False
        elif(password[i] >= '0' and password[i] <= '9'):
            has_number = False
        else:
            for j in special_characters:
                if(j==password[i]):
                    has_special = False
                    break;
        i+=1
    ans = 0
    if(has_lower):
        ans+=1
    if(has_upper):
        ans+=1
    if(has_number):
        ans+=1
    if(has_special):
        ans+=1
    if(n<6):
        ans = max(ans, 6 - n)
    return ans

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())

    password = input()

    answer = minimumNumber(n, password)

    fptr.write(str(answer) + '\n')

    fptr.close()
